fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalProject.java

```diff
@@ -278,4 +278,22 @@ public void computeFd(DataTrait.Builder builder) {
             }
         }
     }
+
+    @Override
+    public Plan replaceExpressions(Map<? extends Expression, ? extends Expression> replaceMap) {
+        List<NamedExpression> newProjections = new ArrayList<>();
+        boolean changed = false;
+        for (NamedExpression expr : getProjects()) {
+            if (replaceMap.containsKey(expr) && replaceMap.get(expr) instanceof NamedExpression) {
```

Comment:
only replace top level expression? the impl is not match the interface's name. should do tree traverse replacement or rename this interface as replaceRootExpression. if replacing only root is expected, we should not add this interface in Plan, because it is not very general